extensions/azure/processors/PutAzureBlobStorage.cpp

```diff
@@ -179,53 +232,43 @@ void PutAzureBlobStorage::onTrigger(const std::shared_ptr<core::ProcessContext>
     return;
   }
 
-  auto connection_string = getConnectionString(context, flow_file);
-  if (connection_string.empty()) {
-    logger_->log_error("Connection string is empty!");
-    session->transfer(flow_file, Failure);
-    return;
-  }
-
-  std::string container_name;
-  if (!context->getProperty(ContainerName, container_name, flow_file) || container_name.empty()) {
-    logger_->log_error("Container Name is invalid or empty!");
-    session->transfer(flow_file, Failure);
-    return;
-  }
-
-  std::string blob_name;
-  if (!context->getProperty(Blob, blob_name, flow_file) || blob_name.empty()) {
-    logger_->log_error("Blob name is invalid or empty!");
+  auto params = buildAzureBlobStorageParameters(context, flow_file);
+  if (!params) {
     session->transfer(flow_file, Failure);
     return;
   }
 
   std::optional<storage::UploadBlobResult> upload_result;
   {
     // TODO(lordgamez): This can be removed after maximum allowed threads are implemented. See https://issues.apache.org/jira/browse/MINIFICPP-1566
+    // When used in multithreaded environment make sure to use the azure_storage_mutex_ to lock the wrapper so the
+    // client is not reset with different configuration while another thread is using it.
     std::lock_guard<std::mutex> lock(azure_storage_mutex_);
-    createAzureStorageClient(connection_string, container_name);
     if (create_container_) {
```

Comment:
onSchedule sets this create_container_ member but onSchedule is not protected by the azure_storage_mutex_
Could that cause unexpected behaviour in multithreaded enviroment?

Reply:
It should not be a problem as onSchedule is only run once and the onTrigger calls that could be running in parallel may only read that variable. The mutex is used for protecting the Azure client when multiple onTrigger calls could reset the client with different parameters while running in parallel.

Reply:
Yeah I see your point, however onSchedule is on the public interface (so in theory anyone could call that) and the overhead from the mutex would be non-existent (onSchedule is rarely called) so I would still prefer if it was protected as well, purely for object oriented principles. But I am not hell bent on that.

Reply:
That's true, in theory anyone could call it, but we usually handle onSchedule to be only run once before onTrigger calls, and we do not handle the possible concurrency in any other processor either. Also as the feature which will remove the mutexes from onTriggers and instead will mark the processors if they can be run in parallel or not is already in progress I would not add additional mutex usages to it.